### ISSUE TITLE
WinRTError: An outgoing call cannot be made since the application is dispatching an input-synchronous call.

### ISSUE THREAD

**Issue by sebagomez** [winjs/winjs#1408]
I'm creating my AppBar programmatically. That AppBar has a Flyout command (secondary) where I have a ToolBar with Commands as shown here:

![capture](https://cloud.githubusercontent.com/assets/1637983/9525766/1af54cc2-4cbc-11e5-8e3d-70588dce243c.PNG)

When I click of one of those commands I get the Message from the subject with the following stack:

gxapp: error: WinRTError: An outgoing call cannot be made since the application is dispatching an input-synchronous call.

   at get exports._KeyboardInfo._extraOccluded (ms-appx://appbartest/libs/WinJS/js/ui.js:38519:17)
   at get exports._KeyboardInfo._visibleDocHeight (ms-appx://appbartest/libs/WinJS/js/ui.js:38537:13)
   at get exports._KeyboardInfo._layoutViewportCoords (ms-appx://appbartest/libs/WinJS/js/ui.js:38613:13)
   at onClose (ms-appx://appbartest/libs/WinJS/js/ui.js:54790:25)
   at Anonymous function (ms-appx://appbartest/libs/WinJS/js/ui.js:43282:25)
   at notifySuccess (ms-appx://appbartest/libs/WinJS/js/base.js:2095:21)
   at state_success_notify.enter (ms-appx://appbartest/libs/WinJS/js/base.js:1777:21)
   at _run (ms-appx://appbartest/libs/WinJS/js/base.js:1993:17)
   at _completed (ms-appx://appbartest/libs/WinJS/js/base.js:1961:13)
   at Signal_complete (ms-appx://appbartest/libs/WinJS/js/base.js:5265:17)
base.js (3009,13)

What am I doing wrong?

EDIT: I got the same behavior with WinJS 4.1 and 4.2

**Comment by jdalton** [comment]
Hi @sebagomez! Can you provide a bit of the code you're working with or a small repro.

**Comment by sebagomez** [comment]
Sure thing, can you give me an email address I can send it to?

**Comment by jdalton** [comment]
Sure. jddalton[at]microsoft[dot]com.

**Comment by sebagomez** [comment]
I've sent an email with an attachment first and as it bounced back uploaded to OneDrive and sent you the link, hope you got it

**Comment by jdalton** [comment]
Thanks. I received the email and will reply on that side.

**Comment by jdalton** [comment]
I've passed your repro to @AmazingJaze.

**Comment by sebagomez** [comment]
I'm glad to passed it to @AmazingJaze and not to @mehJaze :)

**Comment by AmazingJaze** [comment]
@sebagomez  Sorry for the abnormally long delay. I've not been able to build the project that @jdalton sent me. It says that it cannot find the file AppBarTest\package.appxmanifest

Just to sanity check that we have the correct project, can you confirm it is a project titled "AppBarTest" ?

**Comment by sebagomez** [comment]
Here's the link to what I shared to @jdalton , the package.appsmanifest is there
https://onedrive.live.com/redir?resid=7DD6DBC866DE1C9C!445733&authkey=!AGf963Fkpvcm5xQ&ithint=file%2c7z

**Comment by AmazingJaze** [comment]
I tried again using 7zip to unpack and the appxmanifest problem was resolved, however I am unable to repro this behavior. When I click the ToolBar Command I instead get an exception that Genexus.Actions is undefined.

If I look inside of the console I see the following build error:
APPHOST9623: The app couldn’t resolve ms-appx://2794bf8b-91b5-4db3-8bce-7d428813a20f/monaco/app/actions/action.js because of this error: DATA_NOT_AVAILABLE.

If you just create a single html page with the AppBar, Flyout and ToolBar, are you still able to reproduce the original WinRT error?

**Comment by sebagomez** [comment]
Sorry for the delay, will try to reproduce that with a single page.
Thanks

**Comment by nozzlegear** [comment]
I've just run into this same problem today in a Windows 10 app. I have the same setup, an AppBar with a secondary command that crashes when clicked. Rather than showing a fly out, though, mine calls a function that tries to show a message dialog. 

I've tried placing the contents of the called function in a timeout *and* a WinJS promise, but neither of those worked. The only way I could get the app to show the dialog rather than crash was by placing a breakpoint in the called function.

I've uploaded the app to GitHub and created a specific branch for this issue. The readme on that branch has further details and reproduction steps.

Link: [ConvertKitForWindows/WinJS-Issue-1408](https://github.com/nozzlegear/ConvertKitForWindows/tree/WinJS-Issue-1408)
